tsl/src/remote/data_fetcher.h

```diff
@@ -20,17 +20,33 @@ typedef struct DataFetcher DataFetcher;
 
 typedef struct DataFetcherFuncs
```

Comment:
That looks nice, do we need to refactor other fetchers to follow this interface?

Reply:
They can't do anything smart for the new method we're adding, "rescan on changed parameters". COPY can't be parameterized at all, and cursors can't be prepared. So they continue to work as before, only needed some superficial changes.